Package/PackageActions/List.cs

```diff
@@ -71,9 +72,10 @@ protected override int LockedExecute(CancellationToken cancellationToken)
 
             if (Target == null)
                 Target = FileSystemHelper.GetCurrentInstallationDirectory();
-
+            
             HashSet<PackageDef> installed = new Installation(Target).GetPackages().ToHashSet();
-
+            if (checkOs)
+                installed = installed.Where(pkg => pkg.IsOsCompatible(OS)).ToHashSet();
```

Comment:
Why does the checkOs only apply to installed packages and not the packages found in repositories?

Reply:
Because OS was always applied to packages found in repositories:
`packages.AddRange(PackageRepositoryHelpers.GetPackageNameAndVersionFromAllRepos(repositories, new PackageSpecifier("", versionSpec, Architecture, OS)));`

This fix just applies it to installed packages